datadog-oci-orm/metrics-setup/serviceconnector.tf

```diff
@@ -1,27 +1,46 @@
+locals {
+  # Variables to manage connector hubs
+  metrics_compartments    = toset(split(",", var.metrics_compartments))
+  supported_namespaces    = var.metrics_namespaces
+  compartment_tuple       = [for comp in local.metrics_compartments : { compartment = comp, namespaces = tolist(setintersection(local.supported_namespaces, [for ns in data.oci_monitoring_metrics.existing_namespaces[comp].metrics : ns.namespace])) }]
+  # Order by compartment id
+  temp_derived_namespaces = { for tup in local.compartment_tuple : tup.compartment => tup.namespaces if length(tup.namespaces) > 0 }
+  
+  # Filter out the comaprtments having no namespaces to monitor
+  derived_namespaces      = [for key, value in local.temp_derived_namespaces : { compartment = key, namespaces = value }]
+  max_namespaces          = 15
+  # Having large number of compartments
+  single_compartments   = chunklist([for tup in local.derived_namespaces : tup if length(tup.namespaces) > local.max_namespaces], 1)
+  mumtiple_compartments = chunklist([for val in local.derived_namespaces : { compartment = val.compartment, namespaces = val.namespaces } if length(val.namespaces) > 0 && length(val.namespaces) <= local.max_namespaces], 5)
```

Comment:
This condition seems redundant and can be removed : `length(val.namespaces) > 0`